openedx/core/lib/xblock_builtin/xblock_discussion/tests.py

```diff
@@ -0,0 +1,136 @@
+        <discussion
+            url_name="82bb87a2d22240b1adac2dfcc1e7e5e4" xblock-family="xblock.v1"
+            discussion_id="{discussion_id}"
+            discussion_category="{category}"
+            discussion_target="{target}"
+        />
+        """.format(discussion_id=discussion_id, category=category, target=target)
+        node = self._make_xml_node(xblock_xml)
+
+        patched_load_definition_xml.side_effect = Exception("Irrelevant")
+
+        block = DiscussionXBlock.parse_xml(node, self.runtime_mock, self.keys, self.id_gen_mock)
+        self.assertEqual(block.discussion_id, discussion_id)
+        self.assertEqual(block.discussion_category, category)
+        self.assertEqual(block.discussion_target, target)
+
+    @mock.patch(DISCUSSION_XBLOCK_LOCATION + ".load_definition_xml")
+    @ddt.unpack
+    @ddt.data(
+        ("ID1", "Test category", "Test target"),
+        ("ID2", "Some other category", "Some other target")
+    )
+    def test_legacy_export_format(self, discussion_id, category, target, patched_load_definition_xml):
+        """
+        Test that legacy export XML format can be parsed preserving field values
+        """
+        xblock_xml = """
+            <discussion url_name="82bb87a2d22240b1adac2dfcc1e7e5e4" discussion_id="{discussion_id}"/>
+        """.format(discussion_id=discussion_id)
+        xblock_definition_xml = """
+            <discussion discussion_category="{category}" discussion_target="{target}"/>
+        """.format(category=category, target=target)
+        node = self._make_xml_node(xblock_xml)
+        definition_node = self._make_xml_node(xblock_definition_xml)
+
+        patched_load_definition_xml.return_value = (definition_node, "irrelevant")
+
+        block = DiscussionXBlock.parse_xml(node, self.runtime_mock, self.keys, self.id_gen_mock)
+        self.assertEqual(block.discussion_category, category)
+        self.assertEqual(block.discussion_target, target)
+
+    def test_export_default_discussion_id(self):
+        """
+        Test that default discussion_id values are not exported
+        """
+        target_node = etree.Element('dummy')
+
+        block = DiscussionXBlock(self.runtime_mock, scope_ids=self.keys, field_data=DictFieldData({}))
+        discussion_id_field = block.fields['discussion_id']
+
+        # precondition checks - discussion_id does not have a value and uses UNIQUE_ID
+        self.assertEqual(
+            discussion_id_field._get_cached_value(block),  # pylint: disable=protected-access
+            NO_CACHE_VALUE
+        )
+        self.assertEqual(discussion_id_field.default, UNIQUE_ID)
+
+        block.add_xml_to_node(target_node)
+        self.assertEqual(target_node.tag, "discussion")
+        self.assertNotIn("discussion_id", target_node.attrib)
+
+    @ddt.data("jediwannabe", "iddqd", "itisagooddaytodie")
+    def test_export_custom_discussion_id(self, discussion_id):
+        """
+        Test that custom discussion_id values are exported
```

Comment:
@bradenmacdonald Yes - there are no technical blockers to edit `discussion_id` - the only thing is that it is not exposed in editor.